crud/src/main/java/com/redhat/lightblue/assoc/Conjunct.java

```diff
@@ -88,6 +295,39 @@ public QueryExpression getClause() {
         return clause;
     }
 
+    public ConjunctType getConjunctType() {
+        Set<CompositeMetadata> md=getEntities();
+        int n=md.size();
+        if(n<=1) {
+            // If there is at most one entity in the conjunct, then this is a value expression
+            return ConjunctType.value;
+        } else if(n>2) {
+            return ConjunctType.complex;
+        } else {
+            // The problem arises when the expression uses two
+            // entities This could be an expression that relates
+            // values between two entities (in which case, it is a
+            // relation), or this could be an expression that simply
+            // contains values from two entities combines with an OR
+            // (in which case, it is a complex conjunct).
+
+            
+            if(clause instanceof FieldComparisonExpression||
+               clause instanceof ArrayContainsExpression||
+               clause instanceof NaryFieldRelationalExpression) {
+                // These are simple 2-term expressions
+                return ConjunctType.relation;
+            } else {
+                // At this point, clause is one of:
+                //   - ArrayMatchExpression
+                //   - NaryLogicalExpression
+                //   - UnaryLogicalExpression
+                ClauseIterator itr=new ClauseIterator();
+                return itr.iterate(clause).type;
```

Comment:
What about the case where an unexpected Class is passed in. Should an unsupported exception be thrown?